### PR TITLE
[Data] Fix `iter_tensor_batches` benchmarks

### LINKED ISSUE #33830: [data] release test failure: iter_tensor_batches_benchmark_multi_node
### What happened + What you expected to happen

[INFO 2023-03-28 16:42:45,499] log.py: 11  Test iter_tensor_batches_benchmark_multi_node finished after 0.00 seconds. Last logs:
--
  |  
  | Result of case to-tf-16: {'time': 0.09857736999992994}
  | Running case: iter-torch-batches-32
  | /home/ray/anaconda3/lib/python3.7/site-packages/ray/data/_internal/block_batching/util.py:302: RayDeprecationWarning: This API is deprecated and may be removed in future Ray releases. You could suppress this warning by setting env variable PYTHONWARNINGS="ignore::DeprecationWarning"
  | Use get_node_id() instead
  | node_id = ray.get_runtime_context().node_id
  | iter_torch_batches done, block_format: pyarrow num_rows: 16232 num_blocks: 200 num_batches: 508
  | Result of case iter-torch-batches-32: {'time': 77.325716339}
  | Running case: to-tf-32
  | Result of case to-tf-32: {'time': 0.09836827000003723}
  | Running case: iter-torch-batches-prefetch-32-[1, 10]
  | Traceback (most recent call last):
  | File "iter_tensor_batches_benchmark.py", line 158, in <module>
  | run_iter_tensor_batches_benchmark(benchmark, args.data_size_gb)
  | File "iter_tensor_batches_benchmark.py", line 115, in run_iter_tensor_batches_benchmark
  | prefetch_batches=prefetch_batch,
  | File "/tmp/ray/session_2023-03-28_16-27-08_787174_175/runtime_resources/working_dir_files/s3_ray-release-automation-results_working_dirs_iter_tensor_batches_benchmark_multi_node_ydtyxtfhik__anyscale_pkg_1143f914412c62767d0164836282472d/benchmark.py", line 49, in run
  | output_ds = fn(**fn_run_args)
  | TypeError: iter_torch_batches() got an unexpected keyword argument 'prefetch_batches'

### Versions / Dependencies

master

### Reproduction script

https://buildkite.com/ray-project/release-tests-branch/builds/1493#01872a7e-b389-4f24-8a9d-3128cd37ee3d

### Issue Severity

None

### LINKED ISSUE #33851: [data] release test failure: iter_tensor_batches_benchmark_single_node
### What happened + What you expected to happen

ray.exceptions.RayTaskError(FileNotFoundError): ray::_get_read_tasks() (pid=1220, ip=10.0.2.237)
--
  | File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/read_api.py", line 1868, in _get_read_tasks
  | reader = ds.create_reader(**kwargs)
  | File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/datasource/image_datasource.py", line 66, in create_reader
  | self, size=size, mode=mode, include_paths=include_paths, **kwargs
  | File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/datasource/image_datasource.py", line 152, in __init__
  | **reader_args,
  | File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/datasource/file_based_datasource.py", line 396, in __init__
  | ignore_missing_paths=ignore_missing_paths,
  | File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/datasource/file_meta_provider.py", line 175, in expand_paths
  | yield from _expand_paths(paths, filesystem, partitioning, ignore_missing_paths)
  | File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/datasource/file_meta_provider.py", line 408, in _expand_paths
  | yield from _get_file_infos_serial(paths, filesystem, ignore_missing_paths)
  | File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/datasource/file_meta_provider.py", line 435, in _get_file_infos_serial
  | yield from _get_file_infos(path, filesystem, ignore_missing_paths)
  | File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/datasource/file_meta_provider.py", line 507, in _get_file_infos
  | raise FileNotFoundError(path)
  | FileNotFoundError: air-example-data-2/NoneG-image-data-synthetic-raw

### Versions / Dependencies

master

### Reproduction script

https://buildkite.com/ray-project/release-tests-branch/builds/1493#01872a7e-b386-450f-99e5-6461353d82d2

### Issue Severity

None

### PR DESCRIPTION
**Pull request by amogkam**
Closes https://github.com/ray-project/ray/issues/33830
Closes #33851 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/nightly_tests/dataset/iter_tensor_batches_benchmark.py [hunk 1]
```diff
@@ -103,7 +105,7 @@ def run_iter_tensor_batches_benchmark(benchmark: Benchmark, data_size_gb: int):
             batch_size=batch_size,
         )
 
-    prefetch_batches = [1, 10]
+    prefetch_batches = [1, 4]
```
Comment (scottjlee) [inline release/nightly_tests/dataset/iter_tensor_batches_benchmark.py:108]: any reason why are changing this value now?

**Review by scottjlee — APPROVED** [review]

File: release/nightly_tests/dataset/iter_tensor_batches_benchmark.py (hunk 1 above)
Comment (amogkam) [reply release/nightly_tests/dataset/iter_tensor_batches_benchmark.py:108]: Should’ve been changed beforehand

Comment (c21) [reply release/nightly_tests/dataset/iter_tensor_batches_benchmark.py:108]: sorry why we want to decrease the value from 10 to 4 here?

Comment (amogkam) [reply release/nightly_tests/dataset/iter_tensor_batches_benchmark.py:108]: When I had tried this benchmark manually, I didn’t really see much improvement with 10.

I was meaning to change this to 4, but it had just slipped.

But we can keep both, there’s no harm.

**Review by c21 — APPROVED** [review]

**Comment by c21** [comment]
cc @zhe-thoughts to approve, this is to fix CI test failure.

**Review by zhe-thoughts — APPROVED** [review]
Approved for merging into master

**Pull request merged**